### PR TITLE
Add Section for @call

### LINKED ISSUE #7: @call
inline, noinline, tail calling etc

### PR DESCRIPTION
**Pull request by gsquire**
This patch adds a small section regarding the `@call` builtin.

Closes #7

### CODE REVIEW

File: chapter-2.md [hunk 1]
```diff
@@ -741,6 +741,37 @@ test "precision" {
 }
 ```
 
+# Advanced Function Invocation
+Zig offers a `@call` builtin function which gives you more flexibility with which how a function is
+invoked. There are options to enable or disable tail call optimizations, enable or disalbe inlining,
```
Comment (rudedogg) [inline chapter-2.md:746]: Noticed there's a typo with the second "disable" on this line.

Comment (gsquire) [reply chapter-2.md:746]: Good catch, I'll update it.

File: chapter-2.md [hunk 2]
```diff
@@ -741,6 +741,37 @@ test "precision" {
 }
 ```
 
+# Advanced Function Invocation
+Zig offers a `@call` builtin function which gives you more flexibility with which how a function is
```
Comment (matu3ba) [inline chapter-2.md:745]: you can shorten this to what `@call` does and to me its more about controlling what the compiler does. :)

File: chapter-2.md [hunk 3]
```diff
@@ -741,6 +741,37 @@ test "precision" {
...
+invoked. There are options to enable or disable tail call optimizations, enable or disable inlining,
+as well as force compile time generation.
+
+Example:
```
Comment (matu3ba) [inline chapter-2.md:749]: remove `Example:`

File: chapter-2.md [hunk 4]
```diff
@@ -741,6 +741,37 @@ test "precision" {
...
+
+```zig
+const assert = @import("std").debug.assert;
+
+// These functions are effectively the same but are defined and invoked differently.
```
Comment (matu3ba) [inline chapter-2.md:754]: Thats vague. Why are they the same and how invoked differently? Explaining them sounds more understandable to me. Then likely the comments below can also reduced to the key differences.

**Review by matu3ba — COMMENTED** [review]
This can be shortened/made more precise.

**Comment by gsquire** [comment]
I think this has gone pretty stale and I'm sure there would be a few merge conflicts. I'm going to close it but if anyone else wants to pick this up, feel free.
